src/_pytest/python.py

```diff
@@ -1443,6 +1446,16 @@ def idmaker(
     return resolved_ids
 
 
+def _pretty_fixture_path(func) -> str:
+    cwd = Path.cwd()
+    loc = Path(getlocation(func, str(cwd)))
+    prefix = Path("...", "_pytest")
+    try:
+        return str(prefix / loc.relative_to(_PYTEST_DIR))
```

Comment:
How about we opt for relative to site packages if in site packages and/or relative to start dir if sensible

I believe the real bug is that bestrelpath fails to determine when the absolute path is actually shorter than the best relative one

Reply:
Do we have something already to check if a path is in site packages? From what I remember from things like `hunter`, getting a proper site packages path is quite a pain with differences between Python versions, distributions (Debian/Ubuntu and `dist-packages`), etc. etc.

I suppose bestrelpath could show an absolute path if that is shorter indeed. But given that it's used in various places I'm not sure if I want to introduce such a change...

Reply:
Oh: The path from my example above *is* relative and "correct", but I still believe the elided path is nicer for the output.

Reply:
The proposed solution is already a enhancement, let's take my comment as a follow-up

Reply:
Opened #9292 for the follow-up.